### DISCUSSION TITLE
lesson 9 : Test coverage is not showing all the  testing contracts

### DISCUSSION THREAD

**Discussion by 0xkumar** [Cyfrin/foundry-full-course-cu#420] [Q&A] [answered]
The main contract Raffle.sol is not present in the  test coverages althouth i have  tested it

Analysing contracts...
Running tests...
| File                      | % Lines      | % Statements | % Branches    | % Funcs     |
|---------------------------|--------------|--------------|---------------|-------------|
| script/DeployRaffle.s.sol | 0.00% (0/7)  | 0.00% (0/9)  | 100.00% (0/0) | 0.00% (0/1) |
| script/HelperConfig.s.sol | 0.00% (0/7)  | 0.00% (0/10) | 0.00% (0/2)   | 0.00% (0/2) |
| Total                     | 0.00% (0/14) | 0.00% (0/19) | 0.00% (0/2)   | 0.00% (0/3) |

**Comment by shikhar229169** [comment]
Can you share your file structure in your vs code, with the expanded src folder.

**Comment by 0xkumar** [comment]
![Screenshot 2023-07-24 000344](https://github.com/Cyfrin/foundry-full-course-f23/assets/136639853/d9f8b9ad-05ff-48f9-a24f-f50098938745)

**Reply by shikhar229169** [reply]
It seems correct to me, its strange why its not picking it up, can you push it on github, so I can test it on my system for the issue.

**Reply by 0xkumar** [reply]
okay,I will do it now

**Comment by 0xkumar** [comment]
![Screenshot 2023-07-24 004312](https://github.com/Cyfrin/foundry-full-course-f23/assets/136639853/422ac453-d08f-425e-81f2-7d2904bf26de)

**Reply by 0xkumar** [reply]
git doesn't adding this file

**Reply by shikhar229169** [reply]
Yeah, no need to push the lib one, commit it and push to your branch.

**Reply by 0xkumar** [reply]
Could you please check once I just pushed to the github.

**Reply by shikhar229169** [reply]
Sure!

**Reply by shikhar229169** [reply, marked as answer]
The problem is in the Raffle contract, I noticed that it is happening because of this function which is present in Raffle contract:
```solidity
function testing() public {

}
```
Removing it makes Raffle contract appear in the coverage. Maybe its happening because the function is empty.

**Reply by 0xkumar** [reply]
yeah, its working .Thank You
